forem/forem
# Added the logo upload to the admin -> customization -> config -> images section.

## What type of PR is this? (check all applicable)


## Description

This PR adds the ability to upload a Forem instance logo from the admin config section.

If the image is uploaded successfully, the site logo in the header is updated immediately.

https://user-images.githubusercontent.com/833231/145645354-29ea050e-4a2a-4572-8338-0335dae3923e.mp4

If there is an error, it does not change.

https://user-images.githubusercontent.com/833231/145645363-b58261a7-df19-4df3-912f-a1c18b7622ab.mp4

## Related Tickets & Documents

Closes #15618

## QA Instructions, Screenshots, Recordings

1. To test, login as the Forem admin, email: admin@forem.local, password: password
2. Navigate to http://localhost:3000/admin/customization/config
3. Scroll down to the Images sub section under the All Settings section.
4. Click the Show Info button to expand the subsection and scroll to the Logo field.

![CleanShot 2021-12-10 at 17 28 36](https://user-images.githubusercontent.com/833231/145649382-ee56860e-a247-4219-b86c-1f178334b752.png)

5. Select an image to upload. It needs to be a JPEG or PNG. You will not be permitted to select another type of file.
6. Click the Update Settings button
7. The image uploads successfully, a success message appears in the bottom left of the screen and the site header at the top left updates.

### UI accessibility concerns?

Work in the logo uploader already covers a11y issues, but integrating the logo uploader into the admin config section unveiled some existing issues in the admin config. I kept these fixes to the scope of the PR, so for example the _Update Settings_ button visually still says "Update settings", but for assistive technologies, the button is named "Update image settings."

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![spinner gif](https://media.giphy.com/media/KG4PMQ0jyimywxNt8i/giphy.gif)
